mr/src/main/java/org/apache/iceberg/mr/mapred/serde/objectinspector/IcebergObjectInspector.java

```diff
@@ -0,0 +1,118 @@
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspectorFactory;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory;
+import org.apache.hadoop.hive.serde2.typeinfo.PrimitiveTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.TypeUtil;
+import org.apache.iceberg.types.Types;
+
+public final class IcebergObjectInspector extends TypeUtil.SchemaVisitor<ObjectInspector> {
+
+  public static ObjectInspector create(@Nullable Schema schema) {
+    if (schema == null) {
+      return IcebergRecordObjectInspector.empty();
+    }
+
+    return TypeUtil.visit(schema, new IcebergObjectInspector());
+  }
+
+  public static ObjectInspector create(Types.NestedField... fields) {
+    return create(new Schema(fields));
+  }
+
+  @Override
+  public ObjectInspector field(Types.NestedField field, ObjectInspector fieldObjectInspector) {
+    return fieldObjectInspector;
+  }
+
+  @Override
+  public ObjectInspector list(Types.ListType listTypeInfo, ObjectInspector listObjectInspector) {
+    return ObjectInspectorFactory.getStandardListObjectInspector(listObjectInspector);
+  }
+
+  @Override
+  public ObjectInspector map(Types.MapType mapType,
+                             ObjectInspector keyObjectInspector, ObjectInspector valueObjectInspector) {
+    return ObjectInspectorFactory.getStandardMapObjectInspector(keyObjectInspector, valueObjectInspector);
+  }
+
+  @Override
+  public ObjectInspector primitive(Type.PrimitiveType primitiveType) {
+    final PrimitiveTypeInfo primitiveTypeInfo;
+
+    switch (primitiveType.typeId()) {
+      case BINARY:
+        return IcebergBinaryObjectInspector.get();
+      case BOOLEAN:
+        primitiveTypeInfo = TypeInfoFactory.booleanTypeInfo;
+        break;
+      case DATE:
+        return IcebergDateObjectInspector.get();
+      case DECIMAL:
+        Types.DecimalType type = (Types.DecimalType) primitiveType;
+        return IcebergDecimalObjectInspector.get(type.precision(), type.scale());
+      case DOUBLE:
+        primitiveTypeInfo = TypeInfoFactory.doubleTypeInfo;
+        break;
+      case FLOAT:
+        primitiveTypeInfo = TypeInfoFactory.floatTypeInfo;
+        break;
+      case INTEGER:
+        primitiveTypeInfo = TypeInfoFactory.intTypeInfo;
+        break;
+      case LONG:
+        primitiveTypeInfo = TypeInfoFactory.longTypeInfo;
+        break;
+      case STRING:
+        primitiveTypeInfo = TypeInfoFactory.stringTypeInfo;
+        break;
+      case TIMESTAMP:
+        boolean adjustToUTC = ((Types.TimestampType) primitiveType).shouldAdjustToUTC();
+        return IcebergTimestampObjectInspector.get(adjustToUTC);
+
+      case FIXED:
+      case TIME:
+      case UUID:
+      default:
+        throw new IllegalArgumentException(primitiveType.typeId() + " type is not supported");
```

Comment:
Couldn't fixed by read as binary? And UUID as string? And doesn't Hive support time?

Reply:
First two done. As for `TIME` - Hive supports `DATE` and `TIMESTAMP`, I don't know enough about Iceberg's types to comment on how these differ from `TIME` but I'm guessing it doesn't?